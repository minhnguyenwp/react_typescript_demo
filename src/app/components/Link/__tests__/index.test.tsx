import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import { Link } from '../index';
import { themes } from 'styles/theme/themes';
import { MemoryRouter } from 'react-router-dom';

const renderer = createRenderer();

describe('<Link />', () => {
  renderer.render(
    <MemoryRouter>
      <Link to="/test" theme={themes.light}>
        HeaderLink
      </Link>
    </MemoryRouter>,
  );
  const renderedOutput = renderer.getRenderOutput();
  it('should match snapshot', () => {
    expect(renderedOutput).toMatchSnapshot();
  });
});
