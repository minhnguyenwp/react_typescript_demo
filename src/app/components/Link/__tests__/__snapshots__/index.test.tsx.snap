// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should match snapshot 1`] = `
<div>
  <Styled(Component)
    theme={
      Object {
        "background": "rgba(255,255,255,1)",
        "backgroundVariant": "rgba(251,249,249,1)",
        "border": "rgba(58,52,51,0.12)",
        "borderLight": "rgba(58,52,51,0.05)",
        "primary": "rgba(215,113,88,1)",
        "text": "rgba(58,52,51,1)",
        "textSecondary": "rgba(58,52,51,0.7)",
      }
    }
    to="/test"
  >
    HeaderLink
  </Styled(Component)>
</div>
`;
